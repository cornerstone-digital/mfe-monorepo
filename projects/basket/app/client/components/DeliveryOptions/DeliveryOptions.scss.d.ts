export type Styles = {
  header: string
  column: string
}

export type ClassNames = keyof Styles

declare const styles: Styles

export default styles
