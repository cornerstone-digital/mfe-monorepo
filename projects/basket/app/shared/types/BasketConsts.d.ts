export interface BasketConstTypes {
  SIM_TYPE_ESIMONLY: BasketV2.Hardware['simType']
  SIM_TYPE_HYBRID: BasketV2.Hardware['simType']
  SIM_TYPE_PHYSICAL: BasketV2.Hardware['simType']

  PRODUCT_SUB_CLASS_WATCH: BasketV2.Hardware['productSubClass']
  PRODUCT_SUB_CLASS_HANDSET: BasketV2.Hardware['productSubClass']
  PRODUCT_SUB_CLASS_CORE: BasketV2.Hardware['productSubClass']

  PRODUCT_CLASS_HANDSET: BasketV2.Hardware['productClass']
  PRODUCT_CLASS_EHANDSET: BasketV2.Hardware['productClass']
  PRODUCT_CLASS_SIM_CARD: BasketV2.Hardware['productClass']
  PRODUCT_CLASS_DATA_DEVICE: BasketV2.Hardware['productClass']
  PRODUCT_CLASS_ACCESSORY: BasketV2.Hardware['productClass']
  PRODUCT_CLASS_DISCOUNT: BasketV2.Hardware['productClass']

  SERVICE_PRODUCT_CLASS_INSURANCE: BasketV2.Service['productClass']
  SERVICE_PRODUCT_CLASS_SUPER_WIFI: BasketV2.Service['productClass']
  SERVICE_PRODUCT_CLASS_FIXED_LINE: BasketV2.Service['productClass']
  SERVICE_PRODUCT_CLASS_BONUS_DATA: BasketV2.Service['productClass']
  SERVICE_PRODUCT_CLASS_FIXED_BROADBAND_SERVICE: BasketV2.Service['productClass']
  SERVICE_PRODUCT_CLASS_HYBRID_BUNDLES: BasketV2.Service['productClass']

  PLAN_TYPE_SIMO: 'SIMO'
  PLAN_TYPE_WATCH_SIMO: 'WATCH_SIMO'
  PLAN_TYPE_DATA_SIMO: 'DATA_SIMO'
  PLAN_TYPE_HANDSET: 'HANDSET'
  PLAN_TYPE_BINGO: 'BINGO'
  PLAN_TYPE_BROADBAND_FTTH: 'BROADBAND:FTTH'
  PLAN_TYPE_BROADBAND_FTTP: 'BROADBAND:FTTP'
  PLAN_TYPE_BROADBAND_FTTC: 'BROADBAND:FTTC'
  PLAN_TYPE_DATA_DEVICE: 'DATA_DEVICE'
  PLAN_TYPE_ACCESSORY: 'ACCESSORIES'

  PLAN_TYPE_RED_HYBRID: 'RED_HYBRID_ONLINE'

  MP_TYPE_LIMITED_TIME: 'limited_time'

  BUNDLE_DISPLAY_NAME_ESIM: BasketV2.Bundle['bundleClass']
  BUNDLE_TYPE_BASIC: BasketV2.Bundle['bundleClass']
  BUNDLE_TYPE_BASICS_PLAN: BasketV2.Bundle['bundleClass']
  BUNDLE_PAYMENT_TYPE_PRE: BasketV2.Bundle['bundleClass']
  BUNDLE_CLASS_DATA_DEVICE: BasketV2.Bundle['bundleClass']
  BUNDLE_CLASS_DATA_SIMO: BasketV2.Bundle['bundleClass']
  BUNDLE_CLASS_SIMO: BasketV2.Bundle['bundleClass']
  BUNDLE_CLASS_WATCH_SIMO: BasketV2.Bundle['bundleClass']

  ACCOUNT_CATEGORY_BUSINESS: 'BUSINESS'
  ACCOUNT_CATEGORY_CONSUMER: 'CONSUMER'

  ACCOUNT_TYPE_SMALLBUSINESS: 'SMALL-BUSINESS'

  PAYMENT_POSTPAID: 'POSTPAID'
  UPGRADE_ORDER: 'UPGRADE'

  MAKE_VODAFONE: 'VODAFONE'
}
