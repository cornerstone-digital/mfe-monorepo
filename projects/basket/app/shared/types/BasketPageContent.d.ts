declare namespace BasketPageContent {
  export interface PageContent {
    basket: Basket
  }

  export interface Basket {
    vf_MetaDescription: string
    vf_enableGetinTouch: string
    vf_Modules: VfModules
    vf_PageTitle: string
    id: number
    friendlyName: string
    isBreadcrumbDisplayable: string
    vf_TealiumPageName: string
    vf_Robots: string[]
    name: string
    links_tracker: any[]
    category: string
    parents: any[]
  }

  export interface VfModules {
    basket_tradeIn: BasketTradeIn
    notifications: HbbPortfolioRefresh
    messages: HbbPortfolioRefresh
    basket_router_delivery_information: BasketRouterDeliveryInformation
    basket_esim: BasketEsim
    basket_esim_modal: BasketEsimModalClass
    basket_pac_stac_content: BasketEsimModalClass
    hbb_portfolio_refresh: HbbPortfolioRefresh
    lionheart_modal: Lionheart
    lionheart_basket_banner: Lionheart
    popular_esim_devices: PopularEsimDevices
  }

  export interface BasketEsim {
    content: BasketEsimContent
    name: string
    header: string
    id: number
    friendlyName: string
    parents: any[]
  }

  export interface BasketEsimContent {
    basket_keep_current_sim: HbbPortfolioRefresh
    basket_esim_get_sim: BasketEsimDeviceUpgrade
    basket_sim_card: HbbPortfolioRefresh
    basket_esim_info: BasketEsimDeviceUpgrade
    basket_esim_info_compatible: BasketEsimDeviceUpgrade
    basket_esim_get_sim_compatible: BasketEsimDeviceUpgrade
    basket_esim_device_upgrade: BasketEsimDeviceUpgrade
    basket_esim_simo_upgrade: BasketEsimDeviceUpgrade
  }

  export interface BasketEsimDeviceUpgrade {
    type?: Type
    featureLoading?: string
    acceptCTA?: string
    bodyText: string
    name: string
    header: string
    id: number
    friendlyName: string
    parents: any[]
    ctaURL?: string
    declineCTA?: string
  }

  export enum Type {
    Info = 'info',
  }

  export interface HbbPortfolioRefreshContent {
    hbb_portfolio_refersh_1year_apple_tv?: HbbPortfolioRefresh
    hbb_portfolio_refersh_intelligent_wifi?: HbbPortfolioRefresh
    hbb_portfolio_refersh_no_price_rise?: HbbPortfolioRefresh
    hbb_portfolio_refersh_2pound_together_discount?: HbbPortfolioRefresh
    hbb_portfolio_refresh_outofcontract_price_rise?: HbbPortfolioRefresh
    '30041'?: HbbPortfolioRefresh
    freedom_benefits?: BasketRouterDeliveryInformation
    ROAMING_AIRTIME_BENEFIT?: HbbPortfolioRefresh
    '030010'?: HbbPortfolioRefresh
    '030040'?: HbbPortfolioRefresh
    '030030'?: HbbPortfolioRefresh
    '030030_Business'?: HbbPortfolioRefresh
    '030050'?: HbbPortfolioRefresh
    connectivity_plan?: HbbPortfolioRefresh
    number_associated_watch?: HbbPortfolioRefresh
    chosen_connect?: HbbPortfolioRefresh
    MAX_AVAILABLE?: HbbPortfolioRefresh
    MAX_2_MBPS?: HbbPortfolioRefresh
    MAX_10_MBPS?: HbbPortfolioRefresh
    package_footer?: HbbPortfolioRefresh
    RPI_withoutVAT?: HbbPortfolioRefresh
    rpi?: HbbPortfolioRefresh
    SMALL_BUSINESS_BASKET_FOOTER?: HbbPortfolioRefresh
    flexi_upgrade_fee?: HbbPortfolioRefresh
    basket_bingo_subtotal_footer?: HbbPortfolioRefresh
    basket_label_modal_header?: HbbPortfolioRefresh
    basket_label_your_device_plan?: HbbPortfolioRefresh
    basket_label_your_airtime_plan?: HbbPortfolioRefresh
    basket_label_total_package_cost?: HbbPortfolioRefresh
    basket_label_total_phone_cost?: HbbPortfolioRefresh
    basket_label_total_monthly_cost?: HbbPortfolioRefresh
    basket_label_interest_rate?: HbbPortfolioRefresh
    basket_label_plan_type?: HbbPortfolioRefresh
    basket_label_monthly_cost?: HbbPortfolioRefresh
    basket_label_creditor?: HbbPortfolioRefresh
    basket_label_length_of_contract?: HbbPortfolioRefresh
    basket_label_apr?: HbbPortfolioRefresh
    basket_label_total_credit_amount?: HbbPortfolioRefresh
    basket_label_upfront_cost?: HbbPortfolioRefresh
    basket_label_modal_footer?: HbbPortfolioRefresh
    BSKT_INVALID_INPUT_042?: BasketEsimDeviceUpgrade
    MAX_SB_PLANS_EXCEEDED?: BasketEsimDeviceUpgrade
    DEMO_ERROR_MESSAGE?: DemoErrorMessage
    SUBCATEGORY_SMALLBUSINESS_INELIGIBLE_HANDSET?: BasketEsimDeviceUpgrade
    SUBCATEGORY_SOLETRADER_INELIGIBLE_LOAN_WATCH?: BasketEsimDeviceUpgrade
    SUBCATEGORY_SOLETRADER_INELIGIBLE_LOAN_PHONE?: BasketEsimDeviceUpgrade
    SUBCATEGORY_LOAN_NON_LOAN_MIX_WATCH?: BasketEsimDeviceUpgrade
    SUBCATEGORY_LOAN_NONLOAN_MIX_HANDSET?: BasketEsimDeviceUpgrade
    LOAN_DECLINE_CHECKOUT_WATCH_HANDSET?: BasketEsimDeviceUpgrade
    LOAN_DECLINE_CHECKOUT_HANDSET_WATCH?: BasketEsimDeviceUpgrade
    LOAN_DECLINE_CHECKOUT_WATCH_ONLY?: BasketEsimDeviceUpgrade
    LOAN_DECLINE_CHECKOUT_HANDSET_ONLY?: BasketEsimDeviceUpgrade
    Err_Vou_004?: BasketEsimDeviceUpgrade
    Err_Vou_003?: BasketEsimDeviceUpgrade
    Err_Vou_002?: BasketEsimDeviceUpgrade
    BSKT_INVALID_INPUT_037?: BasketEsimDeviceUpgrade
    Err_Vou_001?: BasketEsimDeviceUpgrade
    BSKT_INVALID_INPUT_068?: BasketEsimDeviceUpgrade
    BASKET_LOAN_CAP?: BasketEsimDeviceUpgrade
    BASKET_WATCH_CAP?: BasketEsimDeviceUpgrade
    BASKET_PHONE_CAP?: BasketEsimDeviceUpgrade
    MAXIMUM_LENDING_LIMIT?: BasketEsimDeviceUpgrade
    REPAYMENT_ISSUES?: BasketEsimDeviceUpgrade
    MAXIMUM_CTNS_WITH_CONCURRENT_LOANS?: BasketEsimDeviceUpgrade
    MAXIMUM_LOANS_FOR_CTN?: BasketEsimDeviceUpgrade
    MAXIMUM_ALLOWED_LOANS?: BasketEsimDeviceUpgrade
    MAX_LOAN_LIMIT_REACHED?: BasketEsimDeviceUpgrade
    BSKT_INVALID_INPUT_001?: BasketEsimDeviceUpgrade
    BSKT_INVALID_INPUT_047?: BasketEsimDeviceUpgrade
    basket_airtime_plan?: BasketEsimDeviceUpgrade
    AIRTIME_DEVICE_MISMATCH?: BasketEsimDeviceUpgrade
    LOAN_INVALID_UPFRONT_COST?: BasketEsimDeviceUpgrade
    LOAN_INVALID_DURATION?: BasketEsimDeviceUpgrade
    basket_airtime_plan_length?: BasketEsimDeviceUpgrade
    TRADEIN_QUOTE_EXP?: BasketEsimDeviceUpgrade
    basket_limited_company_partnership?: BasketEsimDeviceUpgrade
  }

  export interface HbbPortfolioRefresh {
    acceptCTA?: string
    name: string
    id: number
    friendlyName: string
    parents: any[]
    bodyText?: string
    content?: HbbPortfolioRefreshContent
  }

  export interface DemoErrorMessage {
    ctaURL: string
    featureNotAvailable: string
    type: Type
    acceptCTA: string
    bodyText: string
    declineCTA: string
    name: string
    header: string
    id: number
    friendlyName: string
    parents: any[]
  }

  export interface BasketRouterDeliveryInformation {
    subTitle?: string
    mainTitle: string
    name: string
    id: number
    friendlyName: string
    parents: any[]
    subLinks?: string[]
  }

  export interface BasketEsimModalClass {
    img: any[]
    moduleSubType: string
    additionalComponents: any[]
    id: number
    friendlyName: string
    heading: string
    name: string
    mainCopy: string
    parents: any[]
  }

  export interface BasketTradeIn {
    unique_code_label: string
    header_2: string
    name: string
    header_1: string
    textContent: string[]
    id: number
    device_label: string
    title: string
    friendlyName: string
    expiry_date_label: string
    parents: any[]
  }

  export interface Lionheart {
    primaryImg: PrimaryImg
    headerText: string
    buttonValue: string
    content: any[]
    name: string
    id: number
    friendlyName: string
    htmlBlock?: string
    parents: any[]
    linkText?: string
  }

  export interface PrimaryImg {
    'Polygon-background-white': PolygonBackgroundWhite
  }

  export interface PolygonBackgroundWhite {
    smallImage_bloblink_: string
    mediumImage_bloblink_: string
    desktopImage_bloblink_: string
    name: string
    id: number
    friendlyName: string
    parents: Parents
  }

  export interface Parents {
    'Image_P:1480076234475': ImageP1480076234475
  }

  export interface ImageP1480076234475 {
    name: string
    id: number
    parents: any[]
  }

  export interface PopularEsimDevices {
    title: string
    content: PopularEsimDevicesContent
    name: string
    header: string
    id: number
    friendlyName: string
    parents: any[]
  }

  export interface PopularEsimDevicesContent {
    apple_list: BasketRouterDeliveryInformation
    samsung_list: BasketRouterDeliveryInformation
    google_list: BasketRouterDeliveryInformation
  }
}
