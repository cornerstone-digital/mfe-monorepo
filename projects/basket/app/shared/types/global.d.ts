import { LoggerheadConfig } from '@cornerstone-digital/loggerhead'

declare global {
  const WEBPACK_ASSET_PREFIX: string
  const shallow: any
  const mount: any
  const snapshot: any
  const BeforeSuite: any
  const If: React.SFC<{ condition: boolean }>
  const For: React.SFC<{ each: string; index: string; of: object }>
  const Choose: React.SFC
  const When: React.SFC<{ condition: boolean }>
  const Otherwise: React.SFC
  interface Window {
    ___: BuildInfo
    VFUK: ClientGlobalVars
    utag: any
    vfukTnt: {
      basket: {
        planBenefitsAB: boolean
        withoutVat: boolean
      }
    }
  }
  namespace NodeJS {
    const shallow: any
    const mount: any
    interface Global {
      document: Document
      window: Window
      navigator: Navigator
      history: History
      location: Location
    }
    interface Window {
      ___: BuildInfo
      VFUK: ClientGlobalVars
    }
  }
  namespace Express {
    interface Request {
      app: Application
    }
  }
}

export interface BuildInfo {
  BUILD_TIME: string
  BRANCH_HASH: string
  BRANCH_NAME: string
  LAST_TAG: string
  APP_VERSION: string
}

export interface FlagConfig {
  id?: string
  group?: string
  key: string
  environment?: string
  project?: string
  enabled: boolean
  flags: FlagConfig[]
}

export interface FeatureFlagConfig {
  enabled: boolean
  flags: FlagConfig[]
}

export interface FinancialInfo {
  fb_phone_cost?: string
  fb_interest_type?: string
  fb_interest_rate?: string
  fb_total_credit_amount?: string
  fb_apr?: string
  fb_length_contract?: string | number
  fb_creditor?: string
}

export interface FufDetails {
  upfront?: string
  monthly?: string
}

export interface ClientGlobalVars {
  env: {
    TEALIUM_ENVIRONMENT: string
    ASSET_URL: string
    CDN_DOMAIN: string
    AUTH_COOKIE_PREFIX: string
    FUT_PROD_WHITELIST_DOMAINS: string
    ENABLE_UDL_LOGS: string
    DALMATIAN_ENVIRONMENT: string
    ENABLE_DATADOG_RUM: string
  }
  vfukTnt: {
    basket: { [key: string]: string }
  }
  basketData: {
    financialInfo?: FinancialInfo[]
    fuf?: FufDetails
  }
}

export interface GlobalConfig {
  isProd?: boolean
  isTest: boolean
  environment: string
  stubApis: boolean
  pathPrefix: string
  http: {
    port: number
    robots: string
    methodWhiteList: string[]
    headers: {
      cache: {
        msToExpiry: number
      }
    }
  }
  cdnDomain: string
  logger: LoggerheadConfig
  validation: {
    bastion: {
      enabled: false
      multiPort: false
    }
  }
  featureFlags: FeatureFlagConfig
  state?: {
    persist?: boolean
  }
  strictMode?: boolean
  dal: {
    versioning: {
      enabled: boolean
      catalogueVersion: string
    }
  }
}
