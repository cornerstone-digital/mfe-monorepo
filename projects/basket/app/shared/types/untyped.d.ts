declare module '@web-core/*'
declare module '@web-shop-core/*'
declare module '@web-cms-core/*'
declare module '@lib-web-analytics/*'
// declare module '@server/*'
declare module '@vfuk/web-shop-basket/*'
declare module '@vfuk/dalmatian/*'
