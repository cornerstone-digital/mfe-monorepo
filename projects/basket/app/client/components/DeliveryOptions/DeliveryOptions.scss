.header {
  display: flex;
}

.column {
  padding-left: $spacing-fixed-1;
  padding-right: $spacing-fixed-1;
  margin-bottom: $spacing-fixed-4;
}
