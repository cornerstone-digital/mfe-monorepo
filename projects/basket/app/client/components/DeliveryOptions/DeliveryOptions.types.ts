export interface DeliveryOptionsProps {
  hasCollection?: boolean
  hasPremiumDelivery?: boolean
  hasStandardDelivery?: boolean
}
