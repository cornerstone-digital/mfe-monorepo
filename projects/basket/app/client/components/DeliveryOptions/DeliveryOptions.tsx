import Container from '@web-core/components/atoms/Container'
import BlockContainer from '@web-core/components/molecules/BlockContainer'
import { Grid, GridColumn, GridRow } from '@vfuk/core-grid'
import Heading from '@web-core/components/atoms/Heading'
import Paragraph from '@web-core/components/atoms/Paragraph'
import Icon from '@web-core/components/atoms/Icon'

import { DeliveryOptionsProps } from './DeliveryOptions.types'

import styles from './DeliveryOptions.scss'

const renderColumn = (icon: string, title: string, content: JSX.Element) => {
  return (
    <GridColumn className={styles.column}>
      <div>
        <div className={styles['header']}>
          <Icon name={icon} marginRight={2} size={3} />
          <Heading size={5} level={5} fontWeight="bold">
            {title}
          </Heading>
        </div>
        {content}
      </div>
    </GridColumn>
  )
}

const DeliveryOptions = ({ hasCollection, hasPremiumDelivery, hasStandardDelivery }: DeliveryOptionsProps) => {
  return (
    <BlockContainer
      heading={{
        text: 'Delivery options',
        size: 5,
        level: 5,
        marginVertical: 1,
        fontWeight: 'extra-bold',
      }}
      collapse={{ collapsable: true, label: 'Hide' }}
      guttering="none"
      marginTop={2}
    >
      <Container gutters="horizontal">
        <Heading size={5} level={5} marginVertical={1} fontWeight="bold">
          Free standard delivery or collection
        </Heading>
        <Paragraph>You can have your order sent to any address in the UK or collect it from a Vodafone store near you.</Paragraph>
      </Container>
      <Grid>
        <GridRow marginTop={4}>
          {hasCollection && renderColumn('store', 'Click and collect', <Paragraph>Collect from your local Vodafone store.</Paragraph>)}
          {hasStandardDelivery &&
            renderColumn(
              'delivery',
              'Standard delivery',
              <Paragraph>
                Order <strong>before 10pm </strong> for free next day delivery.
              </Paragraph>,
            )}
          {hasPremiumDelivery &&
            renderColumn(
              'express-delivery',
              'Premium delivery',
              <Paragraph>Get next working day delivery and choose a time slot that suits.</Paragraph>,
            )}
        </GridRow>
      </Grid>
      <Container>
        <Paragraph>Please choose your delivery option when you reach the checkout.</Paragraph>
      </Container>
    </BlockContainer>
  )
}

DeliveryOptions.defaultProps = {
  hasCollection: false,
  hasPremiumDelivery: false,
  hasStandardDelivery: false,
}

export default DeliveryOptions
