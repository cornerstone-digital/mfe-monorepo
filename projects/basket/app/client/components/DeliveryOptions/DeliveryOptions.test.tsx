import DeliveryOptions from './DeliveryOptions'
import { DeliveryOptionsProps } from './DeliveryOptions.types'

const props: DeliveryOptionsProps = {
  hasCollection: true,
  hasPremiumDelivery: true,
  hasStandardDelivery: true,
}

function getColumnInput(extendProps: DeliveryOptionsProps) {
  const wrapper = shallow(<DeliveryOptions {...extendProps} />)
  const column = wrapper.find({ className: 'column' })
  const icon = column.find('Icon').prop('name')
  const title = column.find('Heading').prop('children')
  return { icon, title }
}

describe('<DeliveryOptions />', () => {
  test('renders collection column', () => {
    const { icon, title } = getColumnInput({ hasCollection: true })
    expect(icon).toEqual('store')
    expect(title).toEqual('Click and collect')
  })
  test('renders standard delivery column', () => {
    const { icon, title } = getColumnInput({ hasStandardDelivery: true })
    expect(icon).toEqual('delivery')
    expect(title).toEqual('Standard delivery')
  })
  test('renders premium delivery column', () => {
    const { icon, title } = getColumnInput({ hasPremiumDelivery: true })
    expect(icon).toEqual('express-delivery')
    expect(title).toEqual('Premium delivery')
  })
  test('renders without exploding', () => {
    const wrapper = shallow(<DeliveryOptions {...props} />)
    expect(snapshot(wrapper)).toMatchSnapshot()
  })
})
