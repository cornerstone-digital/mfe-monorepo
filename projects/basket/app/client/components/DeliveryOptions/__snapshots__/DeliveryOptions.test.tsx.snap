// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeliveryOptions /> renders without exploding 1`] = `
<BlockContainer
  appearance="light1"
  borderAppearance="shadow"
  collapse={
    Object {
      "collapsable": true,
      "label": "Hide",
    }
  }
  guttering="none"
  heading={
    Object {
      "fontWeight": "extra-bold",
      "level": 5,
      "marginVertical": 1,
      "size": 5,
      "text": "Delivery options",
    }
  }
  marginTop={2}
  theme={null}
>
  <Container
    fullHeight={false}
    gutters="horizontal"
  >
    <Heading
      appearance="inherit"
      focusable={false}
      fontWeight="bold"
      level={5}
      marginVertical={1}
      size={5}
      theme={null}
    >
      Free standard delivery or collection
    </Heading>
    <Paragraph
      theme={null}
    >
      You can have your order sent to any address in the UK or collect it from a Vodafone store near you.
    </Paragraph>
  </Container>
  <Grid
    center={true}
    maxWidth="full"
  >
    <Grid
      disableFlex={false}
      marginTop={4}
      noGutters={false}
    >
      <Grid
        className="column"
      >
        <div>
          <div
            className="header"
          >
            <Icon
              appearance="inherit"
              marginRight={2}
              name="store"
              responsive={true}
              size={3}
              theme={null}
            />
            <Heading
              appearance="inherit"
              focusable={false}
              fontWeight="bold"
              level={5}
              size={5}
              theme={null}
            >
              Click and collect
            </Heading>
          </div>
          <Paragraph
            theme={null}
          >
            Collect from your local Vodafone store.
          </Paragraph>
        </div>
      </Grid>
      <Grid
        className="column"
      >
        <div>
          <div
            className="header"
          >
            <Icon
              appearance="inherit"
              marginRight={2}
              name="delivery"
              responsive={true}
              size={3}
              theme={null}
            />
            <Heading
              appearance="inherit"
              focusable={false}
              fontWeight="bold"
              level={5}
              size={5}
              theme={null}
            >
              Standard delivery
            </Heading>
          </div>
          <Paragraph
            theme={null}
          >
            Order 
            <strong>
              before 10pm 
            </strong>
             for free next day delivery.
          </Paragraph>
        </div>
      </Grid>
      <Grid
        className="column"
      >
        <div>
          <div
            className="header"
          >
            <Icon
              appearance="inherit"
              marginRight={2}
              name="express-delivery"
              responsive={true}
              size={3}
              theme={null}
            />
            <Heading
              appearance="inherit"
              focusable={false}
              fontWeight="bold"
              level={5}
              size={5}
              theme={null}
            >
              Premium delivery
            </Heading>
          </div>
          <Paragraph
            theme={null}
          >
            Get next working day delivery and choose a time slot that suits.
          </Paragraph>
        </div>
      </Grid>
    </Grid>
  </Grid>
  <Container
    fullHeight={false}
    gutters="all"
  >
    <Paragraph
      theme={null}
    >
      Please choose your delivery option when you reach the checkout.
    </Paragraph>
  </Container>
</BlockContainer>
`;
